pkg/export/otel/metrics.go

```diff
@@ -425,8 +425,10 @@ func newMetricsReporter(
 			llog.Debug("evicting metrics reporter from cache")
 			v.cleanupAllMetricsInstances()
 
-			mr.deleteTracesTargetInfo(v)
-			mr.deleteTargetInfo(v)
+			if !mr.pidTracker.ServiceLive(id) {
+				mr.deleteTracesTargetInfo(v)
+				mr.deleteTargetInfo(v)
+			}
```

Comment:
Do you think we might end up having multiple dangling `target_infos` for services that don't exist anymore?

I imagine:
- We stop reporting metrics for an existing service
- When the cache evicts, this function is executed but the target_infos are not removed because the process still exist
- When the process exists, the `deleteTargetInfo` function is not executed.

Maybe we'd need to create some kind of "cleanupResourceMetrics" function that is invoked either here, or after a signal (e.g. on resource removal)